openmeter/customer/repository/customer.go

```diff
@@ -0,0 +1,254 @@
+package repository
+
+import (
+	"context"
+	"fmt"
+
+	customer_model "github.com/openmeterio/openmeter/openmeter/customer"
+	entdb "github.com/openmeterio/openmeter/openmeter/ent/db"
+	customerdb "github.com/openmeterio/openmeter/openmeter/ent/db/customer"
+	"github.com/openmeterio/openmeter/pkg/clock"
+	"github.com/openmeterio/openmeter/pkg/models"
+	"github.com/openmeterio/openmeter/pkg/pagination"
+)
+
+func (r repository) ListCustomers(ctx context.Context, params customer_model.ListCustomersInput) (pagination.PagedResponse[customer_model.Customer], error) {
+	db := r.client()
+
+	query := db.Customer.Query().
+		Where(customerdb.DeletedAtIsNil()) // Do not return deleted customers
```

Comment:
Let's implement the IncludeDeleted, given that it's defined in the ListCustomersInput